spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -196,6 +207,7 @@ protected void doStart() {
 	 */
 	@Override
 	protected void doStop() {
+		setActive(false);
 	}
```

Comment:
Javadocs for these methods needs polishing. Subclasses MUST call `super` (which is a bit brittle I think).

Reply:
Well, I guess I can move this `active` directly into `start()` and `stop()` impls.

Reply:
Well, I probably may hack and set `running` to true in that `subscribeToPublisher()`, so we will have less confusion with extra flag 😄